### DISCUSSION TITLE
Tree + drag

### DISCUSSION THREAD

**Discussion by omqmo** [taiga-family/taiga-ui#5616] [Q&A] [answered]
Дико не хочется тратить день на выяснения совместимости tree + @angular/sdk/drag, поэтому решил вас спросить -  это возможно?

**Comment by waterplea** [comment, marked as answer]
99% что нет. Можно что-то своё придумать с кастомным шаблоном в айтемах дерева, попробую набросать.

**Reply by omqmo** [reply]
100% ожидал подобный ответ. Просто хотелось указать на вектор развития.

**Reply by omqmo** [reply]
> Можно что-то своё придумать с кастомным шаблоном в айтемах дерева, попробую набросать.

Вы о чем-то таком?

```
<tui-root>
  <ng-container
    *ngFor="let node of nav.tree().children"
    [ngTemplateOutlet]="tree"
    [ngTemplateOutletContext]="{ $implicit: node, level: 0 }">
  </ng-container>

  <ng-template #tree let-data let-level=level>
    <tui-expand [expanded]="data.isExpand">
      <div>{{ level }}</div>
      <ng-template tuiExpandContent>
        <ng-container *ngFor="let child of data.children"
                      [ngTemplateOutlet]="tree"
                      [ngTemplateOutletContext]="{ $implicit: child, level: level + 1 }" />
      </ng-template>
    </tui-expand>
  </ng-template>
</tui-root>
```

Просто, в моем конкретном случаи, expand даже предпочтительней, так как с tree все равно приходилось и template переопределять и content.

**Reply by waterplea** [reply]
Вот такую штуку набросал, не понятно только, почему позиционирование косит у тайлов, надо будет покопаться позже:
https://stackblitz.com/edit/angular-nwcnay

**Reply by waterplea** [reply]
Завёл баг:
https://github.com/taiga-family/taiga-ui/issues/5619
Как поправим можно будет этот пример прилизать маленько и добавить в демо.

**Reply by omqmo** [reply]
Круто! Но стоит сразу заметить, что до полноценного примера нужно еще и родителей перемещать. 
В общем, я жду :)
